src/headers.js

```diff
@@ -112,14 +119,13 @@ class Headers {
   set (name, value) {
     const [normalizedName, normalizedValue] = normalizeAndValidateHeaderValue(name, value)
 
-    let index = this[kHeaders].length
-    for (let i = 0; i < this[kHeaders].length; i += 2) {
-      if (normalizedName === this[kHeaders][i] || this[kHeaders][i] > normalizedName) {
-        index = i
-        break
-      }
+    const i = lowerBound(this[kHeaders], normalizedName)
+    if (this[kHeaders][i] === normalizedName) {
+      this[kHeaders][i + 0] = normalizedName
+      this[kHeaders][i + 1] = normalizedValue
+    } else {
+      this[kHeaders].splice(i, 0, normalizedName, normalizedValue)
     }
-    this[kHeaders].splice(index, 0, normalizedName, normalizedValue)
```

Comment:
I'm not sure the current implementation here works? What happens if you call `set` on a name that already exists? Should it be inserted or replaced?

Reply:
Either way I think we are missing a test.